engine/flink/management/periodic/src/main/scala/pl/touk/nussknacker/engine/management/periodic/PeriodicProcessService.scala

```diff
@@ -242,8 +242,6 @@ class PeriodicProcessService(
             status
           ) && deployment.state.status != PeriodicProcessDeploymentStatus.Finished =>
         markFinished(deployment, processState).needsReschedule(value = true)
-      case None if deployment.state.status == PeriodicProcessDeploymentStatus.Deployed =>
```

Comment:
I think by removing this case we would introduce the risk of the job staying in the deployed status indefinitely, even though it may never actually be deployed to Flink, or it may finish or fail and disappear from the web console after the Nussknacker designer has exited.

Perhaps we should consider introducing a new status - deploying? We would wait for a running job to appear within, e.g. 10 minutes. If the job does not appear, we would mark it as failed.

WDYT?

Reply:
That's good idea. But I would propose to expand it in another PR.